### PR TITLE
And rename length check

### LINKED ISSUE #2855: RD doesn't enforce or validate name lengths
VBA has a hard limit on legal identifier lengths:

 - Module name must be <= 31 characters
 - Other names (procedure, variable, constant, parameter, etc.) must be <= 255 characters

Failure to validate these can cause the test explorer to hang (similar to #2641).

Refactor/rename should enforce these limits and prevent renaming an identifier to an illegal name.

Any code that creates new VBA code (e.g. extract method, introduce local variable, etc.) should validate these limits as well.

Under no circumstance should Rubberduck *create* an invalid identifier, and under no circumstance should Rubberduck attempt to execute code that prevents the code from compiling.

### PR DESCRIPTION
**Pull request by Hosch250**
Closes #2855

### CODE REVIEW

File: RetailCoder.VBE/UI/Refactorings/Rename/RenameViewModel.cs [hunk 1]
```diff
@@ -72,7 +72,8 @@ public bool IsValidName
                 return !(NewName.Equals(Target.IdentifierName, StringComparison.InvariantCultureIgnoreCase)) &&
                        char.IsLetter(NewName.FirstOrDefault()) &&
                        !tokenValues.Contains(NewName, StringComparer.InvariantCultureIgnoreCase) &&
-                       !NewName.Any(c => !char.IsLetterOrDigit(c) && c != '_');
+                       !NewName.Any(c => !char.IsLetterOrDigit(c) && c != '_') &&
+                       NewName.Length <= (Target.DeclarationType.HasFlag(DeclarationType.Module) ? 31 : 255);
```
Comment (Vogel612) [inline RetailCoder.VBE/UI/Refactorings/Rename/RenameViewModel.cs]: magic numbers?

Comment (Hosch250) [reply RetailCoder.VBE/UI/Refactorings/Rename/RenameViewModel.cs]: Yup.

Comment (retailcoder) [reply RetailCoder.VBE/UI/Refactorings/Rename/RenameViewModel.cs]: They could be public constants in `Declaration`, no?  `MaxLengthModuleName` and `MaxLengthMemberName`?

Comment (Hosch250) [reply RetailCoder.VBE/UI/Refactorings/Rename/RenameViewModel.cs]: Cool.

**Pull request merged**